src/trigger_settings.cpp

```diff
@@ -112,11 +124,32 @@ TriggerSettings::TriggerSettings(std::shared_ptr<GenericAdc> adc,
 	on_cmb_extern_condition_currentIndexChanged(0);
 	ui->cmb_analog_extern->setCurrentIndex(0);
 	on_cmb_analog_extern_currentIndexChanged(0);
+
+	if(trigger->hasExternalTriggerIn())
+		ui->cmb_extern_src->addItem("External Trigger In");
+	if(trigger->hasCrossInstrumentTrigger())
+		ui->cmb_extern_src->addItem("Logic Analyzer");
+
+	if(trigger->hasExternalTriggerOut()){
+		for(auto val : externalTriggerOutMapping)
```

Comment:
The items in the `externalTriggerOutMapping` could be accessed by constant reference : `const auto &val`

Reply:
fixed